code/modules/admin/gimmick/wizard_rings.dm

```diff
@@ -0,0 +1,280 @@
+		ability_path = /datum/targetable/spell/rathens
+
+	// shockwave
+	// 	name = "ring of rathens shockwave"
+	// 	desc = ""
+	// 	icon_state = "// shockwave"
+	// 	ability_path = /datum/targetable/spell/shockwave
+
+	spell_shield
+		name = "ring of protection"
+		desc = "The jewel in this ring has a small field around it."
+		icon_state = "spell_shield"
+		ability_path = /datum/targetable/spell/magshield
+
+	warp
+		name = "ring of warp"
+		desc = "The jewel set in this ring looks like it's moving, somehow."
+		icon_state = "warp"
+		ability_path = /datum/targetable/spell/warp
+
+	forcewall
+		name = "ring of forcewall"
+		desc = "The colorful cube on this ring seems to repel matter around it."
+		icon_state = "forcewall"
+		ability_path = /datum/targetable/spell/forcewall
+
+	doppelganger
+		name = "ring of doppelganger"
+		desc = "Inside the jewels on this ring, you see yourself inspecting the jewels on this ring."
+		icon_state = "doppelganger"
+		ability_path = /datum/targetable/spell/doppelganger
+
+	polymorph
+		name = "ring of polymorph"
+		desc = "The texture of this ring appears to change periodically."
+		icon_state = "polymorph"
+		ability_path = /datum/targetable/spell/animal
+
+	bull_charge
+		name = "ring of the charging bull"
+		desc = "The bull on this ring appears to be breathing."
+		icon_state = "bull_charge"
+		ability_path = /datum/targetable/spell/bullcharge
+
+	pandemonium
+		name = "ring of pandemonium"
+		desc = "Looking at this ring makes you feel confused."
+		icon_state = "pandemonium"
+		ability_path = /datum/targetable/spell/pandemonium
+
+	golem
+		name = "ring of golem"
+		desc = "The jewel set in this ring appears to be filled with a liquid."
+		icon_state = "golem"
+		ability_path = /datum/targetable/spell/golem
+
+	sticks_to_snakes
+		name = "ring of sticks to snakes"
+		desc = "This ring occasionally makes rattling noises."
+		icon_state = "sticks_to_snakes"
+		ability_path = /datum/targetable/spell/stickstosnakes
+
+	soulguard
+		name = "ring of soulguard"
+		desc = "Looking at the jewel of this ring makes you feel safe."
+		icon_state = "soulguard"
+		ability_path = null
+
+		equipped(var/mob/user, var/slot)
+			..()
+			if (isliving(user))
+				var/mob/living/L = user
+				L.spell_soulguard = 1
+
+		unequipped(var/mob/user)
+			..()
+			if (isliving(user))
+				var/mob/living/L = user
+				L.spell_soulguard = 0
+
+	random_type
+		//Doesn't have these spells. no ring for em: kill, shockwave, and candy_ring. (last one isn't actually a spell)
+
+		New(loc)
+			// var/list/L = childrentypesof(/datum/targetable/spell)
+			// L -= /datum/targetable/spell/kill
+			// L -= /datum/targetable/spell/shockwave
+			// ability_path = pick(L)
+			..()
+			//I'm lazy again.
+			var/list/L = childrentypesof(/obj/item/clothing/gloves/ring/wizard)
+			L -= /obj/item/clothing/gloves/ring/wizard/random_type
+			if (locate(/obj/item/clothing/gloves/ring/wizard/random_type) in L)
+				message_admins("WRONG WRONG WRONG")
```

Comment:
Potential leftover from local testing?

Reply:
Ah, yep my mistake. Idk why I even wanted to check that because it so obviously would never get hit, and never did in my testing.